diesel-wasm-sqlite/src/connection/raw.rs

```diff
@@ -84,45 +90,60 @@ impl RawConnection {
         flags as i32
     }
 
-    /* possible to implement this, but would need to fill in the missing wa-sqlite functions
-    pub(super) fn serialize(&mut self) -> SerializedDatabase {
+    /// Serializes the database from sqlite to be stored by the user/client.
+    pub(super) fn serialize(&self) -> SerializedDatabase {
+        let sqlite3 = crate::get_sqlite_unchecked();
+        let wasm = sqlite3.inner().wasm();
+
+        let p_size = wasm.pstack().alloc(std::mem::size_of::<i64>() as u32);
+        let data_ptr = sqlite3.sqlite3_serialize(&self.internal_connection, "main", &p_size, 0);
+        if data_ptr.is_null() {
+            panic!("Serialization failed");
+        }
+
+        let len = p_size.as_f64().unwrap() as usize;
+        let mut data = vec![0; len as usize];
+
         unsafe {
-            let mut size: ffi::sqlite3_int64 = 0;
-            let data_ptr = ffi::sqlite3_serialize(
-                self.internal_connection.as_ptr(),
-                std::ptr::null(),
-                &mut size as *mut _,
-                0,
-            );
-            SerializedDatabase::new(data_ptr, size as usize)
+            ffi::raw_copy_from_sqlite(data_ptr, len as u32, data.as_mut_slice());
```

Comment:
since we're copying this data from sqlite, we can probably call the `sqlite3_free` on the `data_ptr` returned from `sqlite3_serialize` after copying, so that sqlite doesn't carry this data around anymore

Reply:
I was freeing it, but I was running into out of bounds issues. I believe it's running into a race condition. I'll play around with it to see if I can get that fixed.

Reply:
Oh, I realized I was freeing from the wrong memory space. It's fixed now.